cmd/monaco/deploy/internal/clientset/clientset.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package clientset
+
+import (
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/cmd/monaco/dynatrace"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/deploy"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/manifest"
+)
+
+func NewEnvironmentClients(environments manifest.Environments, dryRun bool) (deploy.EnvironmentClients, error) {
+	clients := make(deploy.EnvironmentClients, len(environments))
+	for _, env := range environments {
+		clientSet, err := NewClientSet(env, dryRun)
+		if err != nil {
+			return deploy.EnvironmentClients{}, err
```

Comment:
Why not return `nil, err` here?

Reply:
As I've just copied these functions out of the general deploy file, I didn't consider improving anything tbh. 
And even if, I'm not sure if a nil map over a zero-initialized one in this error case has any impact?